code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,700 @@
+					shake_animation(4)
+					M.Weaken(3)
+					return
+				else
+					src.visible_message(SPAN_WARNING("[M] almost trips on \the [src]!"), SPAN_WARNING("You almost trip on \the [src]!"))
+					shake_animation(2)
+					return
+		else
+			if(M.m_intent == M_RUN && M.a_intent == I_HURT)
+				src.visible_message(SPAN_WARNING("[M] bumps into \the [src], knocking it over!"), SPAN_WARNING("You bump into \the [src], knocking it over!"))
+				do_crash()
+	return ..()
 
+/obj/machinery/iv_drip/update_icon()
+	cut_overlays()
 	if(beaker)
+		add_overlay("beaker[tipped ? "_tipped" : ""]")
 		var/datum/reagents/reagents = beaker.reagents
 		if(reagents?.total_volume)
-			var/image/filling = image('icons/obj/iv_drip.dmi', src, "reagent")
-
+			var/image/filling = image('icons/obj/iv_drip.dmi', src, "[tipped ? "tipped_" : ""]reagent")
 			var/percent = round((reagents.total_volume / beaker.volume) * 100)
 			switch(percent)
-				if(0 to 9)		filling.icon_state = "reagent0"
-				if(10 to 24) 	filling.icon_state = "reagent10"
-				if(25 to 49)	filling.icon_state = "reagent25"
-				if(50 to 74)	filling.icon_state = "reagent50"
-				if(75 to 79)	filling.icon_state = "reagent75"
-				if(80 to 90)	filling.icon_state = "reagent80"
-				if(91 to INFINITY)	filling.icon_state = "reagent100"
+				if(0 to 9)		filling.icon_state = "[tipped ? "tipped_" : ""]reagent0"
+				if(10 to 24) 	filling.icon_state = "[tipped ? "tipped_" : ""]reagent10"
+				if(25 to 49)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent25"
+				if(50 to 74)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent50"
+				if(75 to 79)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent75"
+				if(80 to 90)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent80"
+				if(91 to INFINITY)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent100"
 
 			var/reagent_color = reagents.get_color()
 			filling.icon += reagent_color
 			add_overlay(filling)
+		if(attached)
+			add_overlay("iv_in[tipped ? "_tipped" : ""]")
+			if(mode)
+				add_overlay("light_green[tipped ? "_tipped" : ""]")
+			else
+				add_overlay("light_red[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("iv_out[tipped ? "_tipped" : ""]")
+	if(tank)
+		if(istype(tank, /obj/item/tank/oxygen))
+			tank_type = "oxy"
+		else if(istype(tank, /obj/item/tank/anesthetic))
+			tank_type = "anest"
+		else if(istype(tank, /obj/item/tank/phoron))
+			tank_type = "phoron"
+		else
+			tank_type = "other"
+		add_overlay("tank_[tank_type][tipped ? "_tipped" : ""]")
+		update_gauge()
+	if(breath_mask)
+		if(breather)
+			add_overlay("mask_on[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("mask_off[tipped ? "_tipped" : ""]")
+		if(epp_active)
+			add_overlay("light_blue[tipped ? "_tipped" : ""]")
+
+/obj/machinery/iv_drip/proc/update_gauge()
+	var/gauge_pressure = 0
+	var/last_gauge_pressure
+	if(tank.air_contents)
+		gauge_pressure = tank.air_contents.return_pressure()
+		if(gauge_pressure > 1015) // tanks.dm TANK_IDEAL_PRESSURE 1015
```

Comment:
```suggestion
		if(gauge_pressure > TANK_IDEAL_PRESSURE )
```

Just use the actual define here, it isn't undefined in tanks.dm.

Reply:
The problem is that it's saying it's undefined when I try to compile in VSCode

Reply:
It's a define, it isn't a var of the tank object. 

All a define does is, when it compiles, the compiler finds every instance of `TANK_IDEAL_PRESSURE` in the code and pretends you wrote `1015`.

Reply:
That's the thing, my VSCode isn't compiling when I change it to the ``TANK_IDEAL_PRESSURE`` and keeps insisting it's an undefined var

I don't know if it's because the define set in the ``code/game/objects/items/weapons/tanks/tanks.dm`` itself over a proper ``code/__defines/tanks.dm`` file

Reply:
So, turns out the ``.../machinery/iv_drip.dm`` initializes earlier than the ``.../tanks/tanks.dm``. One of the fixes I could do is merely move the defines in the``.../tanks/tanks.dm`` to the ``.../__defines/misc.dm``, which allows my VSCode to compile it without having to create a new define file, or messing with the atmos.dm file

Reply:
> So, turns out the `.../machinery/iv_drip.dm` initializes earlier than the `.../tanks/tanks.dm`. One of the fixes I could do is merely move the defines in the`.../tanks/tanks.dm` to the `.../__defines/misc.dm`, which allows my VSCode to compile it without having to create a new define file, or messing with the atmos.dm file

That works.